datafusion/common/src/config.rs

```diff
@@ -3455,4 +3456,55 @@ mod tests {
         let parsed_metadata = table_config.parquet.key_value_metadata;
         assert_eq!(parsed_metadata.get("key_dupe"), Some(&Some("B".into())));
     }
+    #[cfg(feature = "parquet")]
+    #[test]
+    fn test_parquet_writer_version_validation() {
+        use crate::{config::ConfigOptions, parquet_config::DFWriterVersion};
+
+        let mut config = ConfigOptions::default();
+
+        // Valid values should work
+        config
+            .set("datafusion.execution.parquet.writer_version", "1.0")
+            .unwrap();
+        assert_eq!(
+            config.execution.parquet.writer_version,
+            DFWriterVersion::V1_0
+        );
+
+        config
+            .set("datafusion.execution.parquet.writer_version", "2.0")
+            .unwrap();
+        assert_eq!(
+            config.execution.parquet.writer_version,
+            DFWriterVersion::V2_0
+        );
+
+        // Invalid value should error immediately at SET time
+        let result = config.set("datafusion.execution.parquet.writer_version", "3.0");
+        assert!(
+            result.is_err(),
+            "Setting invalid writer_version '3.0' should fail at SET time"
+        );
+        let error_msg = result.unwrap_err().to_string();
+        assert!(
+            error_msg.contains("writer version")
+                || error_msg.contains("1.0")
+                || error_msg.contains("2.0"),
+            "Error message should mention valid writer version values. Got: {error_msg}"
+        );
```

Comment:
Can we collapse this into a single assert? It feels weird to have two separate asserts; and having the second assert do 3 `contains` chained by `or` is not ideal, we should just assert the entire error string